tls.tf

```diff
@@ -23,6 +23,7 @@
 resource "tls_private_key" "root" {
   algorithm = "RSA"
   rsa_bits  = "2048"
+  count = "${local.should_manage_tls}"
```

Comment:
Nit: `count` should always be the first argument in the list with a full blank newline between it and the remaining arguments.